### DISCUSSION TITLE
Datagrid editable

### DISCUSSION THREAD

**Discussion by f4n0** [microsoft/fluentui-blazor#1181] [Ideas]
It would be awesome if the datagrid can handle (by design):
- Insert new row
- Edit row
- Delete Row

Am I wrong?

**Comment by dvoituron** [comment]
I think it's better to enable these features on panels external to the grid, and not allow the grid to manage this natively. 
This avoids performance and accessibility problems, for example.

The simplest thing to do is to create a column containing an **Edit** or **Delete** icon and display the appropriate window.
As Microsoft does in Azure, for example.

![peek](https://github.com/microsoft/fluentui-blazor/assets/8350694/3cc5909d-8e96-4e71-ad0f-723ffba6a886)
